cpp/src/arrow/acero/exec_plan.h

```diff
@@ -555,6 +565,34 @@ struct ARROW_ACERO_EXPORT QueryOptions {
   ///
   /// If set then the number of names must equal the number of output columns
   std::vector<std::string> field_names;
+
+  /// Various compute functions and acero internals will type pun array
+  /// buffers from uint8_t* to some kind of value type (e.g. we might
+  /// cast to int32_t* to add two int32 arrays)
+  ///
+  /// If the buffer is poorly algined (e.g. an int32 array is not aligned
+  /// on a 4-byte boundary) then this is technically undefined behavior.
+  /// However, most modern compilers and CPUs are fairly tolerant of this
+  /// behavior and nothing bad (beyond a small hit to performance) is likely
+  /// to happen.
+  ///
+  /// Note that this only applies to source buffers.  All buffers allocated interally
+  /// by Acero will be suitably aligned.
+  ///
+  /// If this field is set to kWarn then Acero will check if any buffers are unaligned
```

Comment:
Perhaps this section of the comment could be broken into comments for the members of `UnalignedBufferHandling`?

Reply:
I don't believe comments on enum members show up in our API docs.  Though, one could argue that no one reads those :)